kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -74,6 +74,24 @@ object Async:
         }
     end runAndBlock
 
+    /** Runs an asynchronous computation with interrupt masking.
+      *
+      * This method executes the given computation in a context where interrupts are not propagated to previous "steps" of the computation.
+      * The returned computation can still be interrupted, but the interruption won't affect the masked portion. This is useful for ensuring
+      * that cleanup operations or critical sections complete even if an interrupt occurs.
+      *
+      * @param v
+      *   The computation to run with interrupt masking
+      * @return
+      *   The result of the computation, which can still be interrupted
+      */
+    def mask[E, A: Flat, Ctx](v: => A < (Abort[E] & Async & Ctx))(
```

Comment:
Is there a reason you didn't go with `uninterruptible`?

Reply:
I normally follow the naming and semantics of the Twitter Future for async primitives and it uses `mask` for this. I'm not a fan of `uninterruptible` because seems a mouthful but I'd be ok with renaming if you strongly about it.

Reply:
Let's proceed with `mask` to avoid users confusing the interrupt-ability of this Fiber.